XenAdmin/Wizards/PatchingWizard/AutomatedUpdatesBasePage.cs

```diff
@@ -201,20 +208,24 @@ private void WorkerProgressChanged(object sender, ProgressChangedEventArgs e)
                         {
                             // remove the successful action from the cleanup actions (we are running the cleanup actions in case of failures or if the user cancelled the process, but we shouldn't re-run the actions that have already been run)
                             actionsWorker.CleanupActions.Remove(action);
-
-                            // only increase the progress if the action succeeded
-                            int newVal = progressBar.Value + e.ProgressPercentage / backgroundWorkers.Count;
-                            progressBar.Value = newVal > 100 ? 100 : newVal;
                         }
                     }
                 }
 
-                UpdateStatusTextBox();
+                UpdateStatus();
             }
         }
 
-        private void UpdateStatusTextBox()
+        private void UpdateStatus()
         {
+            // only increase the progress if the action succeeded
+            var newVal = backgroundWorkers.Sum(b => b.PercentComplete) / backgroundWorkers.Count;

```

Comment:
When the page is cancelled, this method is called from WorkerCompleted (which is triggered by the cancellation), but `backgroundWorkers.Count` is 0, due to `backgroundWorkers.Clear()` in PageCancelled(), so the progressBar value cannot be calculated in this case.